casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -58,32 +63,72 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
 
   it should "handle multi-parent blocks correctly when they operate on stdout" ignore effectTest {
     def echoContract(no: Int) = s"""new stdout(`rho:io:stdout`) in { stdout!("Contract $no") }"""
-    val time                  = System.currentTimeMillis()
+    checkNoConflicts("Nil", echoContract(1), echoContract(2))
+  }
+
+  it should "handle multi-parent blocks correctly when they operate on volatile produce/consume pairs" in effectTest {
+    checkNoConflicts(
+      "Nil",
+      """@"hi"!(1) | for (_ <- @"hi") { Nil }""",
+      """@"hi"!(2) | for (_ <- @"hi") { Nil }"""
+    ) >>
+      checkConflicts("@0!(0)", "for (_ <- @0) { 0 }", "for (_ <- @0) { 0 }") >>
+      checkNoConflicts("0", "for (_ <- @0) { 0 } | @0!(1)", "for (_ <- @0) { 0 } | @0!(2)") >>
+      checkConflicts("for (_ <- @0) { 0 }", "@0!(1)", "@0!(2)") >>
+      checkNoConflicts("for (@1 <- @0) { 0 }", "@0!(1)", "for (@2 <- @0) { 0 } | @0!(2)") >>
+      // Cases that are conflicting now but should not once we have more merge support:
+      checkConflicts("0", "@0!(0)", "@0!(0)") >>
+      checkConflicts("0", "for (_ <- @0) { 0 }", "for (_ <- @0) { 0 }")
+  }
+
+  private def checkConflicts(base: String, b1: String, b2: String): Effect[Unit] = {
+    val time = System.currentTimeMillis()
+    val deploys = Vector(
+      ConstructDeploy.sourceDeploy(base, time + 1, accounting.MAX_VALUE),
+      ConstructDeploy.sourceDeploy(b1, time + 2, accounting.MAX_VALUE),
+      ConstructDeploy.sourceDeploy(b2, time + 3, accounting.MAX_VALUE),
+      ConstructDeploy.sourceDeploy("Nil", time + 4, accounting.MAX_VALUE)
+    )
+    diamondConflictCheck(deploys, numberOfParentsForDiamondTip = 1)
+  }
+
+  private def checkNoConflicts(base: String, b1: String, b2: String): Effect[Unit] = {
```

Comment:
we need both our `check` methods to test the dags:
```
Nil
/  \
A  B
\  /
BASE
```
and 
```
Nil
/  \
B  A
\  /
BASE
```
so that we see the mergability relation is symmetric (we cut the number of test cases to specify explicitly by 2 this way)